src/registry/persistent.rs

```diff
@@ -47,8 +48,8 @@ impl Registry for PersistentRegistry {
 
     fn write(&self, mut content: impl Read) -> Result<ContentHash, Self::Error> {
         // Create a temporary file to store the content in.
-        let temp_file_path = self.folder.join(uuid::Uuid::now_v7().to_string());
-        let mut temp_file = File::create(self.folder.join(&temp_file_path))?;
+        let temp_file_path = tempdir()?.path().join(uuid::Uuid::now_v7().to_string());
```

Comment:
If you use a temp dir, you will not be able to rename the file in an atomic operation because it is not in the same file system